### PR TITLE
ARROW-12396: [Python][Docs] Clarify serialization/filesystem docstrings about deprecated status

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Currently the docstring itself says nothing about it, there is only the warning when calling the function.

### CODE REVIEW

File: python/pyarrow/hdfs.py [hunk 1]
```diff
@@ -31,6 +31,10 @@ class HadoopFileSystem(lib.HadoopFileSystem, FileSystem):
     FileSystem interface for HDFS cluster.
 
     See pyarrow.hdfs.connect for full connection details
+
+    .. deprecated:: 2.0
+        `pyarrow.hdfs.HadoopFileSystem` is deprecated,
+        please use `pyarrow.fs.HadoopFileSystem instead.
```
Comment (pitrou) [inline python/pyarrow/hdfs.py:37]: Missing the closing quote here.

File: python/pyarrow/hdfs.py [hunk 2]
```diff
@@ -184,13 +188,19 @@ def _libhdfs_walk_files_dirs(top_path, contents):
...
+    All parameters are optional and should only be set if the defaults need
+    to be overridden.
 
     Authentication should be automatic if the HDFS cluster uses Kerberos.
     However, if a username is specified, then the ticket cache will likely
     be required.
 
+    .. deprecated:: 2.0
+        `pyarrow.hdfs.connect` is deprecated,
+        please use `pyarrow.fs.HadoopFileSystem instead.
```
Comment (pitrou) [inline python/pyarrow/hdfs.py:202]: And here as well.

**Review by pitrou — COMMENTED** [review]
Just FTR, is it intended that you prefixed some of the docstrings with "DEPRECATED" but not all?

**Comment by jorisvandenbossche** [comment]
> is it intended that you prefixed some of the docstrings with "DEPRECATED" but not all?

Ah, for serialization.pxi I just replaced the EXPERIMENTAL with DEPRECATED, but I see that not all of them had that label.

File: python/pyarrow/hdfs.py (hunk 1 above)
Comment (jorisvandenbossche) [reply python/pyarrow/hdfs.py:37]: Thanks!

**Review by pitrou — APPROVED** [review]

**Pull request closed without merging**